core/src/main/kotlin/plugability/extensions.kt

```diff
@@ -96,7 +96,9 @@ class ExtendingDSL(private val pluginClass: String, private val extensionName: S
         ExtensionOrdered(extensionPoint, pluginClass, extensionName, action, block)
 
     infix fun <T : Any> Extension<T>.applyIf(condition: DokkaConfiguration.() -> Boolean): Extension<T> =
-        this.setCondition(condition)
+        this.setCondition(this.condition + condition)
+
+    infix operator fun <T: Any> ((T) -> Boolean).plus(other: (T) -> Boolean): (T) -> Boolean = {this(it) && other(it)}
```

Comment:
If your goal was to make multiple `applyIf`s combine with each other to form the new predicate, this plus operator shouldn't be part of the dsl. As it is right now it will be only confusing for end users. Inline it inside of `applyIf`.

Reply:
Changed.